src/components/Carousel.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useEffect } from "react";
+import {
+  Paper,
+  Typography,
+  Grid,
+  IconButton,
+  Card,
+  CardMedia,
+  CardContent,
+  Box,
+} from "@mui/material";
+import KeyboardArrowLeftIcon from "@mui/icons-material/KeyboardArrowLeft";
+import KeyboardArrowRightIcon from "@mui/icons-material/KeyboardArrowRight";
+import ModalDetails from "./ModalDetailes";
+
+const cardTextStyle = {
+  backgroundColor: "rgba(24, 20, 20, 0.7)",
+  color: "#fff",
+  padding: "8px",
+  position: "absolute",
+  bottom: "0",
+  left: "0",
+  right: "0",
+};
+
+function Carousel({ getFunction, title, genreId, apiFetched }) {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [selectedItem, setSelectedItem] = useState(null);
+  const [itemsData, setItemsData] = useState([]);
+
+  const itemsPerPage = 8;
+  const dynamicNumberOfItems = 32;
+
+  useEffect(() => {
+    if (getFunction) {
+      const items =
+        genreId !== null
+          ? getFunction(dynamicNumberOfItems, genreId)
+          : getFunction(dynamicNumberOfItems);
+      const parsedItems = items.map((jsonString) => JSON.parse(jsonString));
+      setItemsData(parsedItems);
+    } else if (apiFetched) {
+      setItemsData(apiFetched);
+    }
+  }, [getFunction, apiFetched, genreId]);
+  const totalObjects = itemsData.length;
+  const totalPages = Math.ceil(totalObjects / itemsPerPage);
```

Comment:
```js
const totalPages = useMemo(() => Math.ceil(totalObjects / itemsPerPage))
```

explain me why this way and not your way

Reply:
because usememo will optimize the performance by memoizing the result  and it will change only if totalObjects or itemsPerPage will change.
and they probably wont change